library/src/main/java/com/firebase/ui/FirebaseLoginBaseActivity.java

```diff
@@ -102,6 +172,7 @@ protected void onStart() {
             public void onAuthStateChanged(AuthData authData) {
                 if (authData != null) {
                     mChosenProvider = SocialProvider.valueOf(authData.getProvider());
+                    if (mDialog != null) mDialog.dismiss();
```

Comment:
Do we follow the "if it can be a code block, it must be a code block" rule? I do, and would prefer to follow it here too.